SIL.ReleaseTasks/SetReleaseNotesProperty.cs

```diff
@@ -115,13 +128,54 @@ private string ConvertLatestChangelog(int level, int skipUntilLevel = -1)
 								}
 								else
 								{
-									if (bldr.Length > 0)
-										bldr.AppendLine();
+									
+
+									bool endOfFile = false; 
+									bool currentPackage = false; 
+									bool printLine = false; 
 
 									var headerText = currentLine.Substring(levelHeader.Length);
 									if (!headerText.EndsWith(":"))
 										headerText += ":";
-									bldr.AppendLine(headerText);
+
+									//exclude unnecessary headers from changelog file
+									for (int i = _currentIndex + 1; i < _markdownLines.Length - 1; i++)
+									{
+										var line = _markdownLines[i];
+										if (line.StartsWith("-") && FilterEntries)
+										{
+											if (_filterRegex.IsMatch(line) && line.StartsWith($"- [{PackageId}]"))
+											{
+												currentPackage = true;
+												break;
+											}
+											else
+											{
+												if (!_filterRegex.IsMatch(line) && !string.IsNullOrEmpty(line))
+												{
+													printLine = true;
+													break;
+												}
+											}
+										}
+										else if (!_versionRegex.IsMatch(line) && !string.IsNullOrEmpty(line))
+											break;
+
+										if (i == _markdownLines.Length - 1)
+										{
+											endOfFile = true;
+											break;
+										}
+
+									}
+									
+									if (((currentPackage || printLine) && !endOfFile && FilterEntries) || !FilterEntries)
+									{
+										if (bldr.Length > 0)
+											bldr.AppendLine();
+										bldr.AppendLine(headerText);
+									}	
```

Comment:
nit: trailing whitespace

```suggestion
									}
```